pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -445,51 +540,123 @@ func (p *PoolManager) vmWaitingCleanupLook() {
 	for _ = range c {
 		p.poolMutex.Lock()
 
-		configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
-		if err != nil {
-			logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
-			p.poolMutex.Unlock()
+		p.handleStaleLegacyConfigMapEntries(logger)
+
+		p.handleStaleMacEntries(logger)
+
+		p.poolMutex.Unlock()
+	}
+}
+
+func (p *PoolManager) handleStaleMacEntries(parentLogger logr.Logger) error {
+	logger := parentLogger.WithName("handleStaleMacEntries")
+	var macMapChanged bool
+	for macAddress, macEntry := range p.macPoolMap {
+		macMapChanged = false
+		if !macEntry.hasPendingTransaction() {
 			continue
 		}
+		isEntryStale, err := p.isMacTransactionStale(macAddress)
+		if err == nil && isEntryStale {
+			macMapChanged = true
+			logger.Info("entry is stale", "macAddress", macAddress, "vmFullName", macEntry.instanceName, "interfaceName", macEntry.macInstanceKey, "stale TS", macEntry.transactionTimestamp)
+			vm, err := p.getvmInstance(macEntry.instanceName)
+			if err != nil && apierrors.IsNotFound(err) {
+				logger.Info("vm no longer exists or does not have Template. removing mac from pool", "macAddress", macAddress, "entry", macEntry)
+				p.macPoolMap.removeMacEntry(macAddress)
+			} else if err == nil {
+				p.macPoolMap.alignMacEntryAccordingToVmInterface(macAddress, getVirtualMachineInterfaces(vm))
+			} else {
+				return err
+			}
+		}
+	}
+	if macMapChanged {
+		logger.V(1).Info("macMap is updated", "macPoolMap", p.macPoolMap)
+	}
+	return nil
```

Comment:
Don't use a var just for this, flags make your brain go back and forth, just log it wherever is needed.

Reply:
I flaged it cause I want to avoid a "macMap update" log when nothing really was updated (remember this log is reapeating every 3 seconds), but now that I do it separately I don't need it.